cumulus/polkadot-parachain/src/common/parachain/mod.rs

```diff
@@ -0,0 +1,350 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::common::{
+	command::NodeCommandRunner, BuildImportQueue as BuildImportQueueT, BuildRpcExtensions,
+	BuildSelectChain, ConstructNodeRuntimeApi, NodeBackend, NodeSpec, NodeSpecProvider,
+};
+use cumulus_client_cli::CollatorOptions;
+use cumulus_client_consensus_common::ParachainBlockImport as TParachainBlockImport;
+use cumulus_client_service::{
+	build_network, build_relay_chain_interface, prepare_node_config, start_relay_chain_tasks,
+	BuildNetworkParams, CollatorSybilResistance, DARecoveryProfile, StartRelayChainTasksParams,
+};
+use cumulus_primitives_core::ParaId;
+use cumulus_relay_chain_interface::{OverseerHandle, RelayChainInterface};
+use frame_support::DefaultNoBound;
+use futures::prelude::*;
+use parachains_common::{Block, Hash};
+use polkadot_primitives::CollatorPair;
+use prometheus_endpoint::Registry;
+use sc_consensus::ImportQueue;
+use sc_executor::WasmExecutor;
+use sc_network::{config::FullNetworkConfiguration, service::traits::NetworkBackend, NetworkBlock};
+use sc_service::{Configuration, TFullBackend, TFullClient, TaskManager};
+use sc_sysinfo::HwBench;
+use sc_telemetry::TelemetryHandle;
+use sc_transaction_pool::FullPool;
+use sp_keystore::KeystorePtr;
+use std::{marker::PhantomData, pin::Pin, sync::Arc, time::Duration};
+
+pub mod aura;
+pub mod rpc;
+
+#[cfg(not(feature = "runtime-benchmarks"))]
+pub type HostFunctions = cumulus_client_service::ParachainHostFunctions;
+
+#[cfg(feature = "runtime-benchmarks")]
```

Comment:
Please keep an eye on the end goal that everything to do with benchmark should ideally be removed from this and all other nodes someday. 

I think I've asked this in another PR, but have likely missed the answer 🙈: @ggwpez can you chime in, if there are any functional features that we now miss in the omni-bencher, before we can kiss and say goodbye to the benchmark subcommand here?

Reply:
Sorry, I know this went unaddressed for a while. I hoped we'll discuss it async after merging the PRs. I added it to https://github.com/paritytech/polkadot-sdk/issues/4966 and also will ping @ggwpez

Reply:
Yea we should delete the benchmarking sub-commands here. Omni bencher will be extended for all of them.

Reply:
We can do this after https://github.com/paritytech/polkadot-sdk/pull/4405 is merged. Keeping track of it in #4966